chop/stochastic.py

```diff
@@ -237,17 +270,20 @@ def step(self, step_size=None, closure=None):
                     step_size = 1. / (state['step'] + 1.)
                 else:
                     step_size = self.lr
-                lmo_res, _ = self.lmo(-p.grad, p)
+                lmo_res, _ = self.lmo[idx](-p.grad, p)
                 normalized_grad = lmo_res + p
-                new_p = self.prox(p + step_size * normalized_grad)
+                new_p = self.prox[idx](p + step_size * normalized_grad)
                 state['certificate'] = torch.norm((p - new_p) / step_size)
                 p.copy_(new_p)
+                idx += 1
         return loss
 
 
 class S3CM(Optimizer):
-    """Stochastic Three Composite Minimization (S3CM)
-    Cf
+    """
+    Stochastic Three Composite Minimization (S3CM)
+    ==============================================
+    See
```

Comment:
You can format references with the directive References:  and sphinx formats it very nicely. See for example the docstring of minimize_proximal_gradient in https://github.com/openopt/copt/blob/master/copt/proximal_gradient.py